lib/timers.js

```diff
@@ -418,6 +404,27 @@ function processImmediate() {
 }
 
 
+function tryOnImmediate(immediate, queue) {
+  var threw = true;
+  try {
+    immediate._onImmediate();
+    threw = false;
+  } finally {
+    if (threw === true) {
```

Comment:
I'm not sure I follow, how would that change?

The `try - finally` will still catch the exception?